transaction/_transaction.py

```diff
@@ -138,6 +138,8 @@ def user(self):
     def user(self, v):
         if v is not None:
             self._user = text_or_warn(v)
+        else:
```

Comment:
Sigh, in retrospect, my suggestion was dumb, because blinders. Sorry.

IMO, we should drop the None checks altogether (or assert the value is str or bytes).

(But doing nothing more is OK w me too.)